### ISSUE TITLE
[BUG] allowedHeadingLevels not working.

### ISSUE THREAD

**Issue by gaki2** [mdx-editor/editor#559]
If you want to ask for support or request features, [sponsor the project](https://github.com/sponsors/petyosi) and [contact me over email](mailto:petyo@mdxeditor.dev).

- [o] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [o] I have read the documentation and cannot find an answer.

**Describe the bug**

After setting allowedHeadingLevels [1,2], but `### + enter` shortcut makes h3 tag. also h4, h5, h6.

```tsx

export default function InitializedMDXEditor({
  editorRef,
  ...props
}: { editorRef: ForwardedRef<MDXEditorMethods> | null } & MDXEditorProps) {
  return (
    <MDXEditor
      plugins={[
        headingsPlugin({
          allowedHeadingLevels: [1, 2],
        }),
        markdownShortcutPlugin(),
      ]}
      {...props}
      ref={editorRef}
    />
  );
}

```

**Reproduction**
Start from https://codesandbox.io/p/sandbox/mdx-editor-base-q8s7zr?file=/src/App.tsx to reproduce the problem so that I can observe the issue on my side and make sure that the fix reliably addresses it.

**To Reproduce**

**Expected behavior**

I set allowedHeadingLevels = [1,2] but ### or #### or ##### or ###### still working.

**Screenshots**

https://github.com/user-attachments/assets/7e5f5c99-789f-4bc4-a682-fe760de621c5

**Desktop (please complete the following information):**
 - MacOs, tauri

**Comment by petyosi** [comment]
Agreed. One more point towards moving the markdown shortcuts from lexical into the mdxeditor code. Misfortunate.
